test/e2e/deployment.go

```diff
@@ -394,7 +380,8 @@ func (d *Deployment) waitForEnvoyDaemonSetUpdated() error {
 		if err := d.client.Get(context.TODO(), client.ObjectKeyFromObject(d.EnvoyDaemonSet), tempDS); err != nil {
 			return false, err
 		}
-		return tempDS.Status.NumberAvailable > 0 &&
+		return tempDS.Status.ObservedGeneration == tempDS.Generation &&
```

Comment:
Added in the observed generation vs. generation check which I think should mostly prevent any race conditions here.